ext/mysqlnd/mysqlnd_result.c

```diff
@@ -843,7 +843,7 @@ MYSQLND_METHOD(mysqlnd_result_unbuffered, fetch_row)(MYSQLND_RES * result, void
 
 					if (flags & MYSQLND_FETCH_NUM) {
 						Z_TRY_ADDREF_P(data);
-						zend_hash_next_index_insert(row_ht, data);
+						zend_hash_index_add(row_ht, i, data);
```

Comment:
Same as on the other PR, these calls can now fail and thus leak. Can you please also add test coverage for the case where the column name and index clash?

Reply:
Oh, thanks! This should be fixed now.